rqd/rqd/rqutil.py

```diff
@@ -145,7 +145,7 @@ def getHostIp():
 
 def getHostname():
     """Returns the machine's fully qualified domain name"""
-    if platform.system() == "Linux":
+    if platform.system() in ("Linux", "Windows"):
         if rqd.rqconstants.RQD_USE_IP_AS_HOSTNAME:
             return getHostIp()
         else:
```

Comment:
I think on Windows we need to handle the case if `RQD_USE_IP_AS_HOSTNAME` is false.
The `socket.gethostbyaddr(socket.gethostname())` call fails for me on Windows and can simply be `socket.gethostname()`. 

Looking at the history here, I think there are some Linux environments that needed the `gethostbyaddr` line.

Reply:
What version of Python are you running? `socket.gethostbyaddr(socket.gethostname())` works for me on the machines I've tested (two). I'm running Python3.7

The problem with using socket.gethostname() for me was it was providing the hostname but was unable to resolve it to an IP. This is why I needed it to use IP as the hostname. 

See attached screenshot 
![socket](https://user-images.githubusercontent.com/49570383/89332917-0ac71f80-d649-11ea-99be-1da4b49e56d9.PNG)

Reply:
I'm on Python 3.8.5 and the issue is that `socket.gethostbyaddr()` is unable to find the host returned by `socket.gethostname()` ...
![image](https://user-images.githubusercontent.com/2965608/89346198-7fa45480-d65d-11ea-8319-adf39b19d8f7.png)

I'm wondering if this might be a local configuration problem on my box though?

Reply:
Interesting! That also makes me wonder if it's a configuration problem on my box as well.

Quick question for you, I don't think it should make a difference but are you running Python in Powershell? Do you get a different result using cmd.exe? I'll test this myself shortly

Reply:
Also another question -- seems like this could potentially a tricky issue according to [this](https://stackoverflow.com/questions/36644073/socket-gethostbyaddr-returns-unknown-host). Would you feel adding a try/except here makes logical sense as a solution?

Reply:
My local macOS machine also cannot resolve my machine's hostname:

```
>>> socket.gethostbyaddr(socket.gethostname())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
socket.gaierror: [Errno 8] nodename nor servname provided, or not known
>>> socket.gethostbyname(socket.gethostname())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
socket.gaierror: [Errno 8] nodename nor servname provided, or not known
```

 I agree that this all seems highly dependent on the user's/host's particular configuration.

My vote here would be something like:
- Drop the `platform.system()` check.
- Always attempt to obey the `RQD_USE_IP_AS_HOSTNAME` setting.
- As @Crash-GHaun suggested, wrap it all in a try/except block that falls back to using the hostname, and logs a warning to let the user know what's going on.

Reply:
Totally agree with @bcipriano . Thanks.

Reply:
Updated code to reflect our conversation. Tested on Windows. Please review. Thanks!